homeassistant/components/ourgroceries/todo.py

```diff
@@ -0,0 +1,118 @@
+"""A todo platform for OurGroceries."""
+
+import asyncio
+
+from homeassistant.components.todo import (
+    TodoItem,
+    TodoItemStatus,
+    TodoListEntity,
+    TodoListEntityFeature,
+)
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.core import HomeAssistant, callback
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+from homeassistant.helpers.update_coordinator import CoordinatorEntity
+
+from .const import DOMAIN
+from .coordinator import OurGroceriesDataUpdateCoordinator
+
+
+async def async_setup_entry(
+    hass: HomeAssistant, entry: ConfigEntry, async_add_entities: AddEntitiesCallback
+) -> None:
+    """Set up the OurGroceries todo platform config entry."""
+    coordinator: OurGroceriesDataUpdateCoordinator = hass.data[DOMAIN][entry.entry_id]
+    async_add_entities(
+        OurGroceriesTodoListEntity(coordinator, sl["id"], sl["name"])
+        for sl in coordinator.lists
+    )
+
+
+class OurGroceriesTodoListEntity(
+    CoordinatorEntity[OurGroceriesDataUpdateCoordinator], TodoListEntity
+):
+    """An OurGroceries TodoListEntity."""
+
+    _attr_has_entity_name = True
+    _attr_supported_features = (
+        TodoListEntityFeature.CREATE_TODO_ITEM
+        | TodoListEntityFeature.UPDATE_TODO_ITEM
+        | TodoListEntityFeature.DELETE_TODO_ITEM
+    )
+
+    def __init__(
+        self,
+        coordinator: OurGroceriesDataUpdateCoordinator,
+        list_id: str,
+        list_name: str,
+    ) -> None:
+        """Initialize TodoistTodoListEntity."""
+        super().__init__(coordinator=coordinator)
+        self._list_id = list_id
+        self._attr_unique_id = list_id
+        self._attr_name = list_name
+
+    @callback
+    def _handle_coordinator_update(self) -> None:
+        """Handle updated data from the coordinator."""
+        if self.coordinator.data is None:
+            self._attr_todo_items = None
+        else:
+
+            def _completion_status(item):
```

Comment:
This can be a module level function or entity method instead of a nested function.